feature-libs/quote/components/details/vendor-contact/quote-details-vendor-contact.component.ts

```diff
@@ -5,31 +5,75 @@
  */
 
 import { Component } from '@angular/core';
-import { QuoteFacade } from '@spartacus/quote/root';
 import { EventService } from '@spartacus/core';
-import { ICON_TYPE, MessagingConfigs } from '@spartacus/storefront';
+import {
+  Comment,
+  QuoteDetailsReloadQueryEvent,
+  QuoteFacade,
+} from '@spartacus/quote/root';
+import {
+  ICON_TYPE,
+  MessageEvent,
+  MessagingConfigs,
+} from '@spartacus/storefront';
+import { Observable } from 'rxjs';
+import { map, take } from 'rxjs/operators';
 
 @Component({
   selector: 'cx-quote-details-vendor-contact',
   templateUrl: './quote-details-vendor-contact.component.html',
 })
 export class QuoteDetailsVendorContactComponent {
-  quoteDetails$ = this.quoteFacade.getQuoteDetailsQueryState();
   showVendorContact = true;
   iconTypes = ICON_TYPE;
-  vendorplaceHolder: string = 'Vendor Contact Component';
+
+  quoteDetails$ = this.quoteFacade.getQuoteDetails();
+  messageEvents$: Observable<Array<MessageEvent>> = this.prepareMessageEvents();
 
   messagingConfigs: MessagingConfigs = this.prepareMessagingConfigs();
   constructor(
     protected quoteFacade: QuoteFacade,
     protected eventService: EventService
   ) {}
-  onSend() {
-    //TODO CHHI signature is event: { message: string }
+
+  onSend(event: { message: string }, code: string) {
+    this.quoteFacade
+      .addQuoteComment(code, { text: event.message })
+      .pipe(take(1))
+      .subscribe(() =>
+        this.eventService.dispatch({}, QuoteDetailsReloadQueryEvent)
+      );
   }
+
   protected prepareMessagingConfigs(): MessagingConfigs {
     return {
-      charactersLimit: 20,
+      charactersLimit: 1000,
+      displayAddMessageSection: this.quoteDetails$.pipe(
+        map((quote) => quote.isEditable)
```

Comment:
the comment section will be displayed if EDIT is an allowed action and if it is filtered out in the UI (no edit button present) Otherwise you have to click the edit button first. So if the SELLER is currently now allowed to edit quotes in a specific state, he will also not be able to add a comment. This was my understanding of our last discussion.

Reply:
Yes, your understanding was correct. An additional config for SELLER_DRAFT was missing, I added that. It is working fine for that role. The logic that you implemented seems to be fine. (Maybe later on we need to add a configuration for SELLER_APPROVAL related statuses, but that's not directly related to this PR)